astropy/coordinates/baseframe.py

```diff
@@ -811,8 +814,23 @@ def represent_as(self, base, s='base', in_frame_units=False):
                                       for comp in diff.components)
                     for comp, new_attr_unit in zip(diff.components,
                                                    new_attrs['units']):
-                        if (new_attr_unit and
-                                hasattr(self._data.differentials['s'], comp)):
+                        # Some special-casing to treat a situation where the
```

Comment:
This has to be done because of cases like 
```python
c1 = coord.ICRS(ra=37.4*u.deg, dec=-55.8*u.deg, radial_velocity=105.7*u.km/u.s)
c1.pm_ra_cosdec
```
In `__getattr__`, the representation / differential gets re-represented to the default frame classes, in this case `SphericalRepresentation` and `SphericalCosLatDifferential`. After some discussion in another PR, we decided that the Differentials will follow the behavior of the other classes - that they will keep the vestigial attributes even if they weren't set initially. One other thing we *could* do is manually set the unit of any missing components to `dimensionless` here...As it stands, `c1.pm_ra_cosdec` above has units of `km rad / s`...